plexus-compilers/plexus-compiler-eclipse/src/main/java/org/codehaus/plexus/compiler/eclipse/EclipseJavaCompiler.java

```diff
@@ -247,24 +241,7 @@ public CompilerResult performCompile( CompilerConfiguration config )
         }
 
         // Collect sources
-        Set<String> allSources = new HashSet<>();
-        for ( String source : config.getSourceLocations() )
-        {
-            File srcFile = new File( source );
-            if ( srcFile.exists() )
-            {
-                Set<String> ss = getSourceFilesForSourceRoot( config, source );
-                allSources.addAll( ss );
-            }
-        }
-        for ( String extraSrcDir : extraSourceDirs )
-        {
-            File extraDir = new File( extraSrcDir );
-            if ( extraDir.isDirectory() )
-            {
-                addExtraSources( extraDir, allSources );
-            }
-        }
+        List<String> allSources = Arrays.asList( getSourceFiles( config ) );
```

Comment:
So we are back to a list; I basically rolled back #230 because it doesn't make any sense anymore.